tests/test_gencast_with_thermalizer.py

```diff
@@ -0,0 +1,187 @@
+        print("✓ 2x2 grid test passed!")
+    except Exception as e:
+        print(f"✗ 2x2 grid test failed: {e}")
+        raise
+
+    # Test with medium grid (3x3)
+    print("\n=== Testing 3x3 grid ===")
+    batch, nodes, features = 1, 9, 256
+    x = torch.randn(batch * nodes, features)
+
+    try:
+        out = layer(x, t)  # Should auto-infer dimensions and use simple network
+        print(f"Input shape: {x.shape}")
+        print(f"Output shape: {out.shape}")
+        assert out.shape == x.shape
+        assert not torch.isnan(out).any()
+        print("✓ 3x3 grid test passed!")
+    except Exception as e:
+        print(f"✗ 3x3 grid test failed: {e}")
+        raise
+
+    # Test with larger grid (8x8) - should use full UNet
+    print("\n=== Testing 8x8 grid ===")
+    batch, nodes, features = 1, 64, 256
+    x = torch.randn(batch * nodes, features)
+
+    try:
+        out = layer(x, t)  # Should auto-infer dimensions and use full UNet
+        print(f"Input shape: {x.shape}")
+        print(f"Output shape: {out.shape}")
+        assert out.shape == x.shape
+        assert not torch.isnan(out).any()
+        print("✓ 8x8 grid test passed!")
+    except Exception as e:
+        print(f"✗ 8x8 grid test failed: {e}")
+        raise
+
+
+def test_small_grid_integration():
+    """Test the original small grid case"""
+    print("\n=== Testing original 2x2 grid integration ===")
+    lat_lons = [(i // 2, i % 2) for i in range(4)]  # 2x2 grid
+
+    model = GraphWeatherForecaster(
+        lat_lons,
+        use_thermalizer=True,
+        feature_dim=3,
+        aux_dim=0,
+        node_dim=256,
+        num_blocks=1,
+    )
+
+    features = torch.randn(1, len(lat_lons), 3)
+
+    try:
+        pred = model(features, t=50)  # Fixed timestep for reproducibility
+        print(f"Input shape: {features.shape}")
+        print(f"Output shape: {pred.shape}")
+
+        assert not torch.isnan(pred).any()
+        assert torch.isfinite(pred).all()
+        print("✓ Small grid integration test passed!")
+
+    except Exception as e:
+        print(f"✗ Small grid integration test failed: {e}")
+        raise
+
+
+def test_additional_thermalizer():
+    """Additional thermalizer test"""
+    from graph_weather.models.layers.thermalizer import ThermalizerLayer
+
+    batch, nodes, features = 1, 4, 256
+    x = torch.randn(batch * nodes, features)
+    layer = ThermalizerLayer(input_dim=features)
+    print(f"Thermalizer input shape: {x.shape}")
+    t = torch.randint(0, 1000, (1,)).item()
+    print(f"Timestep: {t}")
+
+    try:
+        out = layer(x, t)  # Should auto-infer dimensions
+        print(f"Thermalizer output shape: {out.shape}")
+        assert out.shape == x.shape
+        assert not torch.isnan(out).any()
+        print("Thermalizer standalone test passed!")
+    except Exception as e:
+        print(f"Thermalizer error: {e}")
+        import traceback
+
+        traceback.print_exc()
+        raise
+
+
+if __name__ == "__main__":
```

Comment:
This should be ran from pytest, so remove this section, it should run wiht `pytest .` from the root of the repo and tests should pass from there.